packages/ra-ui-materialui/src/input/CheckboxGroupInput.js

```diff
@@ -91,160 +83,116 @@ const styles = theme =>
+            } catch (e) {
+                // impossible to convert value, e.g. 'abc'
+                newValue = event.target.value;
+            }
+            if (isChecked) {
+                finalFormOnChange([...(value || []), ...[newValue]]);
+            } else {
+                finalFormOnChange(value.filter(v => v != newValue)); // eslint-disable-line eqeqeq
+            }
+            finalFormOnBlur(); // HACK: See https://github.com/final-form/react-final-form/issues/365#issuecomment-515045503
+        },
+        [finalFormOnChange, finalFormOnBlur, value]
+    );
 
-        return (
-            <FormControl
-                className={className}
-                component="fieldset"
-                margin="normal"
-                {...sanitizeRestProps(rest)}
-            >
-                <FormLabel component="legend" className={classes.label}>
-                    <FieldTitle
-                        label={label}
-                        source={source}
-                        resource={resource}
-                        isRequired={isRequired}
+    return (
+        <FormControl
+            component="fieldset"
+            margin="normal"
+            {...sanitizeRestProps(rest)}
+        >
+            <FormLabel component="legend" className={classes.label}>
+                <FieldTitle
+                    label={label}
+                    source={source}
+                    resource={resource}
+                    isRequired={isRequired}
+                />
+            </FormLabel>
+            <FormGroup row>
+                {choices.map(choice => (
+                    <CheckboxGroupInputItem
+                        choice={choice}
+                        id={id}
+                        onChange={handleCheck}
+                        options={options}
+                        optionText={optionText}
+                        optionValue={optionValue}
+                        translateChoice={translateChoice}
+                        value={value}
                     />
-                </FormLabel>
-                <FormGroup row>{choices.map(this.renderCheckbox)}</FormGroup>
-                {touched && error && (
-                    <FormHelperText error>{error}</FormHelperText>
-                )}
-                {helperText && <FormHelperText>{helperText}</FormHelperText>}
-            </FormControl>
-        );
-    }
-}
+                ))}
+            </FormGroup>
+            {touched && error && <FormHelperText error>{error}</FormHelperText>}
+            {helperText && <FormHelperText>{helperText}</FormHelperText>}
+        </FormControl>
+    );
+};
 
 CheckboxGroupInput.propTypes = {
     choices: PropTypes.arrayOf(PropTypes.object),
-    classes: PropTypes.object,
     className: PropTypes.string,
     label: PropTypes.string,
     source: PropTypes.string,
     options: PropTypes.object,
-    id: PropTypes.string,
-    input: PropTypes.shape({
-        onChange: PropTypes.func.isRequired,
```

Comment:
This property isn't available with the new version? Is this a breaking change?
Same for question for the `meta` prop

Reply:
That's because we're using the hooks from final-form